application/plugin_base/item_list.py

```diff
@@ -1655,11 +1673,34 @@ def get_provider(self, path=None):
 		if path is None:
 			return self._current_provider
 
+		# create a local provider if no provider exists yet as fallback in case we are opening an archive
+		if len(self._providers) == 0:
+
+			Provider = self._parent.get_provider_by_protocol('file')
+			result = Provider(self, user.home)
+
+			# cache provider for later use
+			root_path = result.get_root_path(user.home)
+			self._providers[root_path] = result
+			self._current_provider = result
+
 		# check if there is a provider for specified path
 		if path in self._providers:
 			result = self._providers[path]
 
 		else:
+
+			# assuming a path like /home/user/some.zip/subdir1/subdir2
+			# creates intermediate archive provider if necessary
+			p = path
+			while p not in self._providers and p != '/' and '://' not in path:
```

Comment:
Well, am merely pointing out that we should not prevent users from doing something they might want to do. For example opening archives on FTP or Samba shares, which might be a valid thing to do.

I would say that when inheriting path, it should be perfectly fine to inherit providers as well. That doesn't solve the issue of restoring path or manually navigating to one. But it does go around issues with possible racing conditions when updating archive.

Say you have the following case. You inherit path from within the archive. But on the left side you removed a file from archive before inheriting that path on the right. If you create another archive provider on the other panel it will no longer reflect real situation on the left side because archive is updated only when user exits and provider is no longer needed. Admittedly, I didn't implement this all the way as I planned because as you can see it's quite difficult problem to solve but that's the plan.